numba/pythonapi.py

```diff
@@ -868,35 +871,42 @@ def to_native_value(self, obj, typ):
                                           types.float64, types.float32)
                 c64.real = freal
                 c64.imag = fimag
-                return c64._getvalue()
+                return NativeValue(c64._getvalue(), is_error=failed)
             else:
-                return cplx._getvalue()
+                return NativeValue(cplx._getvalue(), is_error=failed)
 
         elif isinstance(typ, types.NPDatetime):
             val = self.extract_np_datetime(obj)
-            return val
+            return NativeValue(val, is_error=c_api_error())
 
         elif isinstance(typ, types.NPTimedelta):
             val = self.extract_np_timedelta(obj)
-            return val
+            return NativeValue(val, is_error=c_api_error())
+
+        elif isinstance(typ, types.Record):
+            buf = self.alloca_buffer()
+            ptr = self.extract_record_data(obj, buf)
+
+            with cgutils.if_unlikely(self.builder,
+                                     cgutils.is_null(self.builder, ptr)):
+                # XXX
```

Comment:
Is this a marker for setting an exception instead of returning the null value as is done at the moment?